test/aave/ReopenStEth.test.ts

```diff
@@ -0,0 +1,318 @@
+      )
+
+      const [_openTxStatus, d] = await executeThroughProxy(
+        system.common.dsProxy.address,
+        {
+          address: system.common.operationExecutor.address,
+          calldata: system.common.operationExecutor.interface.encodeFunctionData('executeOp', [
+            openStrategy.calls,
+            OPERATION_NAMES.aave.OPEN_POSITION,
+          ]),
+        },
+        signer,
+        depositAmount.toString(),
+      )
+      openTxStatus = _openTxStatus
+
+      console.log(`Transaction: ${d.status}`)
+
+      const positionAfterOpen = await strategies.aave.getCurrentStEthEthPosition(
+        { proxyAddress: system.common.dsProxy.address },
+        { ...dependencies },
+      )
+
+      console.log(`After: ${positionAfterOpen.collateral.amount.toString()}`)
+
+      const closeStrategy = await strategies.aave.closeStEth(
+        {
+          stEthAmountLockedInAave: amountToWei(positionAfterOpen.collateral.amount),
+          slippage,
+        },
+        {
+          ...dependencies,
+          position: positionAfterOpen,
+        },
+      )
+
+      const [_closeTxStatus] = await executeThroughProxy(
+        system.common.dsProxy.address,
+        {
+          address: system.common.operationExecutor.address,
+          calldata: system.common.operationExecutor.interface.encodeFunctionData('executeOp', [
+            closeStrategy.calls,
+            OPERATION_NAMES.aave.CLOSE_POSITION,
+          ]),
+        },
+        signer,
+        '0',
+      )
+
+      closeTxStatus = _closeTxStatus
+
+      const currentPosition = await strategies.aave.getCurrentStEthEthPosition(
+        { proxyAddress: system.common.dsProxy.address },
+        { ...dependencies },
+      )
+
+      const reopenStrategy = await strategies.aave.openStEth(
+        {
+          depositAmount,
+          slippage,
+          multiple,
+        },
+        {
+          ...dependencies,
+          currentPosition: currentPosition,
+        },
+      )
+
+      const [_reopenTxStatus] = await executeThroughProxy(
+        system.common.dsProxy.address,
+        {
+          address: system.common.operationExecutor.address,
+          calldata: system.common.operationExecutor.interface.encodeFunctionData('executeOp', [
+            reopenStrategy.calls,
+            OPERATION_NAMES.aave.OPEN_POSITION,
+          ]),
+        },
+        signer,
+        depositAmount.toFixed(0),
+      )
+      reopenTxStatus = _reopenTxStatus
+    })
+
+    it('Open Tx should pass', () => {
+      expect(openTxStatus).to.be.true
+    })
+    it('Close Tx should pass', () => {
+      expect(closeTxStatus).to.be.true
+    })
+    it('Reopen Tx should pass', () => {
+      expect(reopenTxStatus).to.be.true
+    })
+  })
+
+  describe('Should close position with real oneInch', () => {
+    const slippage = new BigNumber(0.1)
+
+    let openTxStatus: boolean
+    let closeTxStatus: boolean
+    let reopenTxStatus: boolean
+
+    before(async function () {
```

Comment:
What's the point of having before when you can skip before and have on `it` test scenario with 3 expectations in it ?

Reply:
It's tricky. `this.skip()` in that example is `describe` so we are skipping all tests in that describe. 
If we will use `() => ` instead of `function() {}` that would not be working.

Reply:
Lol. I wasn't refering to `.skip`. 
```
describe((`Strategy | AAVE `) => {
   it('should reopen a position',m async () => {
   // open positon
    expect(openTxStatus).to.be.true
    // close positon
    expect(closeTxStatus).to.be.true
    // open position
    expect(reopenTxStatus).to.be.true
   })
})
```

Reply:
Also I wouldn't rely on `txStatus` but rather values in the position. It's possible to call a non existing contract or contract with some fallback function implemented and proxy will return successful call but you would have done absolutely nothing. 
I can remember multiple scenarios like this in our application on production

Reply:
👍

Reply:
I will change the tests